core/src/stores/store.rs

```diff
@@ -311,9 +380,41 @@ pub const POSTGRES_TABLES: [&'static str; 11] = [
        status                   TEXT NOT NULL,
        FOREIGN KEY(data_source) REFERENCES data_sources(id)
     );",
+    "-- databases
+    CREATE TABLE IF NOT EXISTS databases (
+       id                   BIGSERIAL PRIMARY KEY,
+       created              BIGINT NOT NULL,
+       database_id          TEXT NOT NULL, -- unique within data source
+       internal_id          TEXT NOT NULL, -- unique globally
+       data_source          BIGINT NOT NULL,
+       name                 TEXT NOT NULL,
+       connection_string    TEXT, -- unused for now
+       FOREIGN KEY(data_source) REFERENCES data_sources(id)
+    );",
+    "-- databases tables
+    CREATE TABLE IF NOT EXISTS databases_tables (
+       id                   BIGSERIAL PRIMARY KEY,
+       created              BIGINT NOT NULL,
+       table_id             TEXT NOT NULL, -- unique within database
+       internal_id          TEXT NOT NULL, -- unique globally
+       database             BIGINT NOT NULL,
+       name                 TEXT NOT NULL,
+       description          TEXT NOT NULL,
+       FOREIGN KEY(database) REFERENCES databases(id)
```

Comment:
If we want to pre-load the database in memory at the we call the DatabaseSchema block that means we have to be able to return the DatabaseSchema without loading the database. SHould we keep it here?

Reply:
Fair, we can have a nullable schema field. It will be kind of a "cache" I guess, do you think we also need a timestamp for when this schema was last inserted ?

Reply:
I guess it will be kept updated by all inserts in a sense so the last update is the last insert no?

Reply:
Yeah I guess that works 👍